lib/m2ts/caption-stream.js

```diff
@@ -144,22 +175,35 @@
       return;
     }
 
-    // parse out CC data packets
-    captionPackets = parseCaptionPackets(event.pts, userData);
+    // parse out CC data packets and save them for later
+    this.captionPackets_ = this.captionPackets_.concat(parseCaptionPackets(event.pts, userData));
+  };
 
-    // send the data to the appropriate field
-    for (i = 0; i < captionPackets.length; i++) {
-      if (captionPackets[i].type === 0) {
-        this.field1_.push(captionPackets[i]);
-      }
+  CaptionStream.prototype.flush = function () {
+    // make sure we actually parsed captions before proceeding
+    if (!this.captionPackets_.length) {
+      this.field1_.flush();
```

Comment:
This is just for completeness - flush on field1_ triggers `done` which triggers `done` on CaptionStream. I think streams should always propagate `done` events to any future stream even if they haven't emitted any data.